### PR TITLE
User update + POC for optimistic locking (NFR)

### PR DESCRIPTION
**Pull request by campingtree**
* Update functionality for user.
* POC for doing optimistic locking. My idea of it working is as follows:
     * Frontend calls /user or /user/detailed to get `etag` header. This header is base64 of `RowVersion` in DB.
     * Call to /update *optionally* asks for `etag` header. If provided, it uses it to check if the entity hasn't yet been updated since last call to /user or /user/detailed. If it has, endpoint returns 409.
     * Frontend handles this by either letting the user overwrite (just calling /update without `etag`) or by calling /user again to get the new `etag`. 
 * Merged migrations into one ( again :) ) because last migrations contained seed data.

### CODE REVIEW

File: src/Mantasflowers.Contracts/User/Request/UpdateUserAddressRequest.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    {
+        public string Country { get; set; }
+
+        public string City { get; set; }
+
+        public string Street { get; set; }
+
+        public string Zipcode { get; set; }
+    }
+}
```
Comment (alwaysintune) [inline src/Mantasflowers.Contracts/User/Request/UpdateUserAddressRequest.cs]: Replace with:
```cs
public class UpdateUserAddressRequest : AddressTemplate { }
```
Can be already found in Common\Templates. Same with UpdateUserContactInfoRequest. Also, with UpdateUser.* responses.

File: src/Mantasflowers.Services/Services/User/UserService.cs [hunk 2]
```diff
@@ -83,26 +85,95 @@ public async Task<string> GetUserUidByGuidAsync(Guid id)
...
+            {
+                throw new FirebaseUidNotFoundException($"No database user matching the {nameof(uid)} could be found");
+            }
+
+            await _fbService.DeleteUserByUidAsync(uid);
+
+            // These entities aren't linked for automatic cascading. Need to delete by hand.
+            var ordersToDelete = user.UserOrders.Select(x => x.Order);
+            var shipmentsToDelete = ordersToDelete.Select(x => x.Shipment);
+            var paymentsToDelete = ordersToDelete.Select(x => x.Payment);
```
Comment (alwaysintune) [inline src/Mantasflowers.Services/Services/User/UserService.cs]: Already submitted Orders should never be deleted.

File: src/Mantasflowers.Contracts/User/Request/UpdateUserAddressRequest.cs (hunk 1 above)
Comment (alwaysintune) [reply src/Mantasflowers.Contracts/User/Request/UpdateUserAddressRequest.cs]: Noticed that namings of request and response classes are missing Post/Get prefixes.

Comment (campingtree) [reply src/Mantasflowers.Contracts/User/Request/UpdateUserAddressRequest.cs]: Those prefixes were never supposed to be there in the first place. Just that some requests started with `Get` :D

File: src/Mantasflowers.Services/Services/User/UserService.cs (hunk 2 above)
Comment (campingtree) [reply src/Mantasflowers.Services/Services/User/UserService.cs]: Why not? If we don't delete them when deleting the user, they will essentially be there forever, with no purpose to serve. Also GDPR pepe be agri.

Comment (alwaysintune) [reply src/Mantasflowers.Services/Services/User/UserService.cs]: Having a hole in an order history would be hard to explain, when paying taxes against fraud. Also, order does not contain any information on a user. You'd have to delete from UserOrder table, though.

Comment (campingtree) [reply src/Mantasflowers.Services/Services/User/UserService.cs]: Makes sense. Will adjust. UserOrder already contains cascading.

**Review by alwaysintune — APPROVED** [review]

**Pull request merged**